dbt/adapters/athena/config.py

```diff
@@ -1,9 +1,108 @@
+        is not defined, it falls back to the default timeout value. If the retrieved timeout value is less than or
+        equal to 0, a ValueError is raised as timeout must be a positive integer.
+
+        Returns:
+            int: The timeout value in seconds.
+
+        Raises:
+            ValueError: If the timeout value is not a positive integer.
+
+        """
+        timeout = self.config.get("timeout", DEFAULT_SPARK_SESSION_TIMEOUT)
+        if not isinstance(timeout, int):
+            raise TypeError("Timeout must be an integer")
+        if timeout <= 0:
+            raise ValueError("Timeout must be a positive integer")
+        LOGGER.debug(f"Setting timeout: {timeout}")
+        return timeout
+
+    def get_polling_interval(self) -> Any:
+        """
+        Get the polling interval for the configuration.
+
+        Returns:
+            Any: The polling interval value.
+
+        Raises:
+            KeyError: If the polling interval is not found in either `self.config`
+                or `self.session_kwargs`.
+        """
+        try:
+            return self.config["polling_interval"]
+        except KeyError:
+            try:
+                return self.session_kwargs["polling_interval"]
+            except KeyError:
+                return DEFAULT_POLLING_INTERVAL
+
+    def set_polling_interval(self) -> float:
+        """
+        Set the polling interval for the configuration.
+
+        Returns:
+            float: The polling interval value.
+
+        Raises:
+            ValueError: If the polling interval is not a positive integer.
+        """
+        polling_interval = self.get_polling_interval()
+        if not (isinstance(polling_interval, float) or isinstance(polling_interval, int)) or polling_interval <= 0:
+            raise ValueError(f"Polling_interval must be a positive number. Got: {polling_interval}")
+        LOGGER.debug(f"Setting polling_interval: {polling_interval}")
+        return float(polling_interval)
+
+    def set_engine_config(self) -> Dict[str, Any]:
+        """Set the engine configuration.
+
+        Returns:
+            Dict[str, Any]: The engine configuration.
+
+        Raises:
+            TypeError: If the engine configuration is not of type dict.
+            KeyError: If the keys of the engine configuration dictionary do not match the expected format.
+        """
+        engine_config = self.config.get("engine_config", DEFAULT_SPARK_ENGINE_CONFIG)
+        if not isinstance(engine_config, dict):
+            raise TypeError("Engine configuration has to be of type dict")
+
+        expected_keys = {"CoordinatorDpuSize", "MaxConcurrentDpus", "DefaultExecutorDpuSize"}
+        if set(engine_config.keys()) != expected_keys:
+            raise KeyError(f"The keys of the dictionary entered do not match the expected format: {expected_keys}")
+
+        if engine_config["MaxConcurrentDpus"] == 1:
```

Comment:
@Avinash-1394 let's put a global variable MAX_CONCURRENT_DPUS on top in case AWS changed in the future.

Reply:
In `dbt/adapters/athena/constants.py` I have added `DEFAULT_SPARK_ENGINE_CONFIG = {"CoordinatorDpuSize": 1, "MaxConcurrentDpus": 2, "DefaultExecutorDpuSize": 1}` which is the smallest configuration with which you can run a spark session. Do you think it would be better to split that into three variables?

Also should I put that global variable in some other place or is that `constants.py` alright?

Reply:
Maybe 3 variables are better in this case. constants.py as for the others seems the right place

Reply:
I have split it into three variables. But when it is overridden in the model it will still need to be passed as a whole dictionary like 

```python
dbt.config(
        materialized="table",
        engine_config={
            "CoordinatorDpuSize": 1,
            "MaxConcurrentDpus": 3,
            "DefaultExecutorDpuSize": 1,
        },
```

Reply:
that's fine.